perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/ResourcesManagerBlImpl.java

```diff
@@ -93,6 +95,12 @@ public Resource getResourceById(PerunSession sess, int id) throws ResourceNotExi
 		return getResourcesManagerImpl().getResourceById(sess, id);
 	}
 
+	@Override
+	public EnrichedResource getEnrichedResourceById(PerunSession sess, int id, List<String> attrNames) throws ResourceNotExistsException {
+		Resource resource = getResourceById(sess, id);
+		return filterOnlyAllowedAttributes(sess, convertToEnrichedResource(sess, resource, attrNames));
```

Comment:
This is not the right place for filtering. On the business logic layer, you need to get always all objects you could have. Filtrations need to be done on entry layer.